rust/connlib/tunnel/src/client.rs

```diff
@@ -89,9 +89,14 @@ impl ClientTunnel {
     }
 
     pub fn set_disabled_resources(&mut self, new_disabled_resources: BTreeSet<ResourceId>) {
+        let old_routes = self.role_state.routes();
         self.role_state
             .set_disabled_resource(new_disabled_resources);
 
+        if HashSet::from_iter(self.role_state.routes()) == HashSet::from_iter(old_routes) {
```

Comment:
@thomaseizinger I'd like your input on this, we can either make `set_disabled_resources` return something to indicate that the routes must be updated or we do it from here, that's not tested for now.

Reply:
This is fine for now. We are gonna redesign these events later.